BGL/include/CGAL/boost/graph/shortest_path.h

```diff
@@ -0,0 +1,160 @@
+* Computes the shortest path between two vertices in a graph `g`
+*
+*@tparam Graph a model of the concept `HalfedgeListGraph`
+* @param vs source vertex
+* @param vt target vertex
+* @param g the graph
+* @param halfedge_sequence_oit the output iterator holding the output sequence
+* of halfedges that form the shortest path from `vs` to `vt` on `g`
+* @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+* ** edge_weight_map : @todo deal with input mesh with no internal pmap.
+*                       default in boost is `get(boost::edge_weight, mesh)`
+*
+*/
+template<typename Graph,
+         typename OutputIterator,
+         typename NamedParameters = parameters::Default_named_parameters>
+OutputIterator shortest_path_between_two_vertices(
+  const typename boost::graph_traits<Graph>::vertex_descriptor vs,//source
+  const typename boost::graph_traits<Graph>::vertex_descriptor vt,//target
+  const Graph& g,
+  OutputIterator halfedge_sequence_oit,
+  const NamedParameters& np = parameters::default_values())
+{
+  using vertex_descriptor = typename boost::graph_traits<Graph>::vertex_descriptor;
+  using halfedge_descriptor = typename boost::graph_traits<Graph>::halfedge_descriptor;
+
+  using Pred_umap = std::unordered_map<vertex_descriptor, vertex_descriptor>;
+  using Pred_pmap = boost::associative_property_map<Pred_umap>;
+
+  using parameters::get_parameter;
+  using parameters::choose_parameter;
+
+  const auto w_map = choose_parameter(get_parameter(np, internal_np::edge_weight),
+                                      get(boost::edge_weight, g));
+
+  Pred_umap predecessor;
+  Pred_pmap pred_pmap(predecessor);
+
+  internal::Stop_at_target_Dijkstra_visitor<Graph> vis;
+  vis.destination_vd = vt;
+  try
+  {
+    boost::dijkstra_shortest_paths(g, vs,
+      boost::predecessor_map(pred_pmap)
+      .visitor(vis)
+      .weight_map(w_map));
+  }
+  catch (const internal::Dijkstra_end_exception& ){}
+
+  // Walk back from target to source and collect vertices along the way
+  struct vertex_on_path
+  {
+    vertex_descriptor vertex;
+    bool is_constrained;
+  };
+
+  std::vector<vertex_descriptor> constrained_vertices = { vs };
+  vertex_descriptor t = vt;
+  std::vector<vertex_on_path> path;
+  do
+  {
+    const bool is_new_vertex = (constrained_vertices.end()
+      == std::find(constrained_vertices.begin(), constrained_vertices.end(), t));
+
+    vertex_on_path vop;
+    vop.vertex = t;
+    vop.is_constrained = !is_new_vertex;
+    path.push_back(vop);
+
+    t = get(pred_pmap, t);
+  }
+  while (t != vs);
+
+  // Add the last vertex
+  vertex_on_path vop;
+  vop.vertex = constrained_vertices.back();
+  vop.is_constrained = true;
+  path.push_back(vop);
+
+  // Display path
+  for (auto path_it = path.begin(); path_it != path.end() - 1; ++path_it)
+  {
+    const std::pair<halfedge_descriptor, bool>
+      h = halfedge((path_it + 1)->vertex, path_it->vertex, g);
```

Comment:
Can't we obtain directly the halfedges when running Dijkstra, instead of  having a sequence of vertices?

Reply:
I think we can by implementing a `DijkstraVisitor::edge_relaxed(e)`  and store the edge in a property map of the target vertex. See the documentation [here](https://www.boost.org/doc/libs/1_42_0/libs/graph/doc/dijkstra_shortest_paths.html):  _`vis.edge_relaxed(e, g)` is invoked on edge `(u,v)` if `d[u] + w(u,v) < d[v]`. The edge `(u,v)` that participated in the last relaxation for vertex `v` is an edge in the shortest paths tree._